Integrations/RiskSense/RiskSense.py

```diff
@@ -0,0 +1,1975 @@
+    """
+    hr = tableToMarkdown('Application Details:', get_app_details_hr(app_detail_dict),
+                         ['Address', 'Name', 'Network Name', 'Network Type', 'Discovered On', 'Last Found On'])
+
+    findings_distribution = get_findings_distribution_hr(app_detail_dict.get('findingsDistribution'))
+    hr += '\n' + tableToMarkdown(
+        'Findings Distribution:',
+        findings_distribution,
+        ['Total', 'Critical', 'High', 'Medium', 'Low', 'Info'])
+
+    hr += tableToMarkdown('Tag(s) (' + str(len(app_detail_dict.get('tags'))) + '):',
+                          get_tag_details_hr(app_detail_dict.get('tags')),
+                          ['Name', 'Category', 'Description', 'Created', 'Updated'])
+
+    hr += tableToMarkdown('Ticket(s) (' + str(len(app_detail_dict.get('tickets'))) + '):',
+                          get_ticket_detail_hr(app_detail_dict.get('tickets')),
+                          ['TicketNumber', 'TicketStatus', 'DeepLink', 'Type', 'ConnectorName'])
+
+    hr += '\n ### Group Details: '
+    hr += '\n Name: ' + app_detail_dict.get('group', {}).get('name', '')
+
+    hr += '\n ### Sources: '
+    hr += '\n Scanner(s): ' + get_source_detail_hr(app_detail_dict.get('sources', {}))
+
+    app_description = app_detail_dict.get('description', '')
+    if app_description:
+        hr += '\n ### Application Description:' + '\n' + app_description
+
+    return hr
+
+
+def get_request_timeout():
+    """
+    Validate and return the request timeout parameter.
+    The parameter must be a positive integer.
+    Default value is set to 60 seconds for API request timeout.
+
+    :params req_timeout: Request timeout value.
+    :return: boolean
+    """
+    try:
+        request_timeout = int(demisto.params().get('request_timeout'))
+        if request_timeout <= 0:
+            raise ValueError
+        return request_timeout
+    except ValueError:
+        return_error('HTTP Request Timeout parameter must be a positive integer.')
+
+
+def get_self_link(resp):
+    """
+    Retrieve self link from response.
+
+    :param resp: JSON response
+    :return: self link
+    """
+    return resp.get('_links', {}).get('self', {}).get('href', '')
+
+
+def fetch_page_details(resp):
+    """
+    Parse total element, page number and total pages from the page.
+
+    :param resp: json response.
+    :return: page details.
+    """
+    total_element = resp.get('page', {}).get('totalElements', 0)
+    page_number = resp.get('page', {}).get('number', 0)
+    total_pages = resp.get('page', {}).get('totalPages', 0)
+
+    return page_number, total_element, total_pages
+
+
+''' REQUESTS FUNCTIONS '''
+
+
+def test_module(client):
+    """
+        Performs basic GET request
+    :param client:
+    :return: None
+    """
+    client_name = demisto.params().get('client_name', '-')
+    resp_json = client.http_request('GET', url_suffix='')
+    clients = resp_json.get('_embedded', {}).get('clients', [])
```

Comment:
better to put this in a try, except as the URL might be faulty

Reply:
@ronykoz if URL is faulty then it will fall under main method's catch exception and will use `return_error` method

Reply:
I understand, but we want to make sure that the error will be clear enough for the user, so he will just have to read it and will know what to search for/do in order to fix it

Reply:
@ronykoz we can do that but, we are getting DemistoException (CommonServerPython) hence won't be able to differentiate the type of exception (in this case ConnectionError). Let us know if we have missed anything.

Reply:
If you wrap this code section with try except you should be able to catch it here, and then return_error with a good msg attached to it

Reply:
okay @ronykoz, will that in `test_module`